src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -2,13 +2,16 @@ package com.github.quillraven.fleks
 
 import com.github.quillraven.fleks.collection.*
 import kotlinx.serialization.Serializable
-import kotlin.jvm.JvmInline
 
 /**
  * An entity of a [world][World]. It represents a unique id.
  */
 @Serializable
-data class Entity(val id: Int, val version: Int = 0)
+data class Entity(val id: Int, val version: Long) {
```

Comment:
I think the change from `value class` to `data class` might be bigger than we initially thought. I am really wondering why the benchmarks are not showing that but maybe because they are too short to trigger the garbage collector (GC).

I now had some time to look into more details and there are several places where Fleks is using `Entity(id)` to "create" an entity. For example in `DefaultEntityProvider.create`, `DefaultEntityProvider.forEach`, `BitArray.toEntityBag` or the entire `MutableEntityBag` class.

Granted, you already optimized it by making `Entity(-1)` a constant `Entity.NONE` but in the other areas we are now always creating a new class instance of an entity instead of working with a primitive Int.

I think this is not a good idea and could cause bad behavior especially on mobile devices like Android which are not as fast and efficient when it comes to GC.

----

One way to keep this approach and mitigate this issue, is to keep the `value class` and encode the version into the entity id, but let's see.

Reply:
Would be great if you can also compare `data class` to `value class` in your memory test. Imo the value class should perform a lot better since it should be just a simple `Int` or `Long` which is more efficient to handle.

Also, if I understood your comment correctly, you anyway were able to pass the entire `Entity` class around without the usage of `BitArray`. If that is the case and if the performance is almost the same then I'd vote to revert to the value class again. My example of the other PR worked fine for me and was doing well until I introduced all the different lookups :D

But let's wait for your findings. I am excited to see the outcome!